### PR TITLE
Support role graph

### PR DESCRIPTION
**Pull request by jjmason**

### CODE REVIEW

File: lib/conjur/api/roles.rb [hunk 1]
```diff
@@ -19,9 +19,32 @@
...
+require 'conjur/graph'
 
 module Conjur
   class API
+    ##
+    # Fetch a digraph (or a forest of digraphs) representing of
+    #   role memberships related transitively to any of a list of roles.
+    #
+    # @param [Array<Conjur::Role, String>] roles the  digraph (or forest thereof) of
+    #   the ancestors and descendants of these roles or role ids will be returned
```
Comment (dividedmind) [inline lib/conjur/api/roles.rb]: BTW, @kgilpin , @jjmason , I was thinking -- what was the rationale for transitive descendant visibility? It seems overly lax and I couldn't think of why it's needed after I added 'skin' visibility to the graph, but I believe I wasn't following the original discussion so I could have missed something. Failing that, maybe we should drop it before it becomes released and entrenched?

File: lib/conjur/graph.rb [hunk 2]
```diff
@@ -0,0 +1,213 @@
+#
+# Copyright (C) 2013 Conjur Inc
```
Comment (dividedmind) [inline lib/conjur/graph.rb]: Jon, pull that old calendar off the wall already ;)

File: lib/conjur/graph.rb [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+# FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+# COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+# IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+# CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+#
+module Conjur
+  # A Graph represents a digraph of role memberships.
+  # (it's not called Digraph because I don't like either of the two possible camel case forms -- jjm)
+  # Currently it just has an edges member, but in future iterations it might support methods for graph
+  # properties and various useful output formats
```
Comment (dividedmind) [inline lib/conjur/graph.rb]: I believe this comment is no longer relevant :)

**Comment by dividedmind** [comment]
Nice, thanks! I like `Graph#to_dot` :)
But in my opinion `#to_png` does not belong in the API gem (or our code in general, insofar as it just calls `dot`, which is much more transparent and convenient to pipe directly using a shell).

File: lib/conjur/api/roles.rb (hunk 1 above)
Comment (kgilpin) [inline lib/conjur/api/roles.rb]: I'm not sure what you mean.

Comment (dividedmind) [inline lib/conjur/api/roles.rb]: We allowed user to see (_transitively_) what roles belong to his. I'm not sure we want that anymore. (That'd be the lower cone on the image I drew the other day.)

Eg. do we want the user to be able to see members of admin if admin is a member of the user?

Comment (kgilpin) [inline lib/conjur/api/roles.rb]: Probably not. But it's not really an api-ruby issue.

**Pull request merged**